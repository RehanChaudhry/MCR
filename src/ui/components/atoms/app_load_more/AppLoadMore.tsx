import { usePreferredTheme } from "hooks";
import React from "react";
import {
  ActivityIndicator,
  StyleProp,
  StyleSheet,
  TextStyle,
  TouchableOpacityProps,
  View
} from "react-native";
import { AppLabel } from "ui/components/atoms/app_label/AppLabel";
import { optimizedMemo } from "ui/components/templates/optimized_memo/optimized_memo";
import { FONT_SIZE } from "config";

export interface AppLoadMoreProps extends TouchableOpacityProps {
  text?: string;
  loaderSize?: number | "small" | "large";
  textStyle?: StyleProp<TextStyle>;
  shouldShowLoadMore?: boolean;
  shouldNotOptimize?: boolean;
}

export const AppLoadMore = optimizedMemo<AppLoadMoreProps>(
  ({
    loaderSize = "small",
    text = "Loading, please wait",
    textStyle,
    shouldShowLoadMore = true
  }) => {
    const theme = usePreferredTheme();
    const view = () => {
      return (
        <View style={style.row}>
          <ActivityIndicator
            size={loaderSize}
            color={theme.themedColors.label}
          />
          <AppLabel
            text={text}
            style={[style.textStyle, textStyle]}
            numberOfLines={3}
          />
        </View>
      );
    };
    return (
      <View style={style.container}>
        {shouldShowLoadMore ? view() : null}
      </View>
    );
  }
);

const style = StyleSheet.create({
  container: {
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "row"
  },
  row: {
    flexDirection: "row",
    justifyContent: "center",
    paddingVertical: 10
  },
  textStyle: {
    paddingLeft: 5,
    fontSize: FONT_SIZE.sm
  }
});
