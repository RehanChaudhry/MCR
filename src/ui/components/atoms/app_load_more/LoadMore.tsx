import { usePreferredTheme } from "hooks";
import React from "react";
import {
  ActivityIndicator,
  StyleProp,
  StyleSheet,
  TextStyle,
  TouchableOpacityProps,
  View
} from "react-native";
import { AppLabel } from "ui/components/atoms/app_label/AppLabel";

export interface AppLoadMoreProps extends TouchableOpacityProps {
  text?: string;
  loaderSize?: number;
  textStyle?: StyleProp<TextStyle>;
  shouldShowLoadMore?: boolean;
}

export const LoadMore = React.memo<AppLoadMoreProps>(
  ({
    loaderSize = 25,
    text = "Loading, please wait",
    textStyle,
    shouldShowLoadMore = true
  }) => {
    const theme = usePreferredTheme();
    const view = () => {
      return (
        <>
          <ActivityIndicator
            size={loaderSize}
            color={theme.themedColors.primaryIconColor}
          />
          <AppLabel
            text={text}
            style={[style.textStyle, textStyle]}
            numberOfLines={3}
          />
        </>
      );
    };
    return (
      <View style={style.container}>
        {shouldShowLoadMore ? view() : null}
      </View>
    );
  }
);

const style = StyleSheet.create({
  container: {
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "row"
  },
  textStyle: {
    paddingLeft: 5
  }
});
