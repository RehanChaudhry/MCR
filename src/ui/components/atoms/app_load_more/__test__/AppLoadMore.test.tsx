import { render } from "@testing-library/react-native";
import React from "react";
import { AppLoadMore } from "ui/components/atoms/app_load_more/AppLoadMore";

test("snapshot testing", () => {
  const rendered = render(<AppLoadMore />).toJSON();
  expect(rendered).toMatchSnapshot();
});
