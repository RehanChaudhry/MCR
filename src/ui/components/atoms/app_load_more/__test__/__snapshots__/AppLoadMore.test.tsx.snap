// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#111827"
    hidesWhenStopped={true}
    size={25}
  />
  <Text
    numberOfLines={3}
    style={
      Array [
        Array [
          Object {
            "fontFamily": "Inter-Regular",
          },
          Object {
            "color": "#111827",
          },
          Object {
            "lineHeight": 35,
          },
          Array [
            Object {
              "fontSize": 25,
              "paddingLeft": 5,
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "Inter-Regular",
        },
      ]
    }
  >
    Loading, please wait
  </Text>
</View>
`;
